File: GED-api/app/models/folder.py
from datetime import datetime
from typing import List, Optional
from sqlalchemy.orm import Mapped, mapped_column, relationship
from sqlalchemy import String, ForeignKey, DateTime, func
from app.core.database import Base

class Folder(Base):
    __tablename__ = "folders"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    name: Mapped[str] = mapped_column(String, index=True, nullable=False)
    parent_id: Mapped[Optional[int]] = mapped_column(ForeignKey("folders.id", ondelete="CASCADE"), index=True)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), onupdate=func.now(), nullable=True)
    created_by_id: Mapped[int] = mapped_column(ForeignKey("users.id"))

    # Relationships
    subfolders: Mapped[List["Folder"]] = relationship("Folder", backref="parent", remote_side="Folder.id")
    documents = relationship("Document", back_populates="folder")

File: GED-front/src/types/folder.ts
export interface Folder {
  id: number
  name: string
  parent_id: number | null
  path_slug: string
  created_at: string
  updated_at: string
  uploaded_by_id: number
  is_deleted: boolean
  subfolders?: Folder[]
}

export interface FolderTree {
  id: number
  name: string
  subfolders: FolderTree[]
}

export interface BreadcrumbItem {
  id: number
  name: string
}
